gateway/ln-gateway/src/lnd.rs

```diff
@@ -18,14 +18,18 @@ use tracing::{error, info, trace};
 use crate::gatewaylnrpc::get_route_hints_response::{RouteHint, RouteHintHop};
 use crate::gatewaylnrpc::intercept_htlc_response::{Action, Cancel, Forward, Settle};
 use crate::gatewaylnrpc::{
-    GetNodeInfoResponse, GetRouteHintsResponse, InterceptHtlcRequest, InterceptHtlcResponse,
-    PayInvoiceRequest, PayInvoiceResponse,
+    EmptyResponse, GetNodeInfoResponse, GetRouteHintsResponse, InterceptHtlcRequest,
+    InterceptHtlcResponse, PayInvoiceRequest, PayInvoiceResponse,
 };
 use crate::lnrpc_client::{ILnRpcClient, RouteHtlcStream, MAX_LIGHTNING_RETRIES};
 use crate::GatewayError;
 
 type HtlcSubscriptionSender = mpsc::Sender<Result<InterceptHtlcRequest, Status>>;
 
+lazy_static::lazy_static! {
```

Comment:
Any reason why this needs to be global? Couldn't we embedded it into some struct?

Reply:
Same question on my side, mutable globals are a big code smell. Given its usage I'd put it into `GatewayLndClient`.

Reply:
Open to suggestions on how to solve this, I'm not a fan of having this be global either.

The reason it cannot be embedded in `GatewayLndClient` right now is because we use multiple lightning clients in different spots. We use one lightning client for handling HTLC interceptions and we use a different lightning client when "completing" those HTLCs.

We could refactor so that each of these operations uses the same client, but that essentially restricts us to a single `GatewayLndClient` that can be used, which functionally is similar to a global variable. Perhaps its cleaner code though.

Reply:
Actually, using `Settleinvoice` and `CancelInvoice` will probably solve this problem

Reply:
@m1sterc001guy, check out #2755 which avails these `invoicesrpc` features you need

Reply:
This has been removed.